modules/light-clients/08-wasm/types/client_state_test.go

```diff
@@ -938,217 +938,137 @@ func (suite *TypesTestSuite) TestVerifyNonMembershipGrandpa() {
+					suite.Require().Equal(path, payload.VerifyNonMembership.Path)
+					suite.Require().Equal(proof, payload.VerifyNonMembership.Proof)
+
+					bz, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					return &wasmvmtypes.Response{Data: bz}, types.DefaultGasUsed, nil
+				})
+			},
+			nil,
+		},
+		{
+			"success: with update client state",
+			func() {
+				suite.mockVM.RegisterSudoCallback(types.VerifyNonMembershipMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, sudoMsg []byte, store wasmvm.KVStore,
+					_ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction,
+				) (*wasmvmtypes.Response, uint64, error) {
+					var payload types.SudoMsg
+					err := json.Unmarshal(sudoMsg, &payload)
+					suite.Require().NoError(err)
+
+					suite.Require().NotNil(payload.VerifyNonMembership)
+					suite.Require().Nil(payload.CheckSubstituteAndUpdateState)
+					suite.Require().Nil(payload.UpdateState)
+					suite.Require().Nil(payload.UpdateStateOnMisbehaviour)
+					suite.Require().Nil(payload.VerifyMembership)
+					suite.Require().Nil(payload.VerifyUpgradeAndUpdateState)
+					suite.Require().Equal(proofHeight, payload.VerifyNonMembership.Height)
+					suite.Require().Equal(path, payload.VerifyNonMembership.Path)
+					suite.Require().Equal(proof, payload.VerifyNonMembership.Proof)
+
+					bz, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					expClientStateBz = []byte("client-state-data")
+					store.Set(host.ClientStateKey(), expClientStateBz)
+
+					return &wasmvmtypes.Response{Data: bz}, types.DefaultGasUsed, nil
+				})
+			},
+			nil,
+		},
+		{
+			"wasm vm returns invalid proof error",
+			func() {
+				proof = []byte("invalid proof")
+
+				suite.mockVM.RegisterSudoCallback(types.VerifyNonMembershipMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, _ wasmvm.KVStore,
+					_ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction,
+				) (*wasmvmtypes.Response, uint64, error) {
+					return nil, types.DefaultGasUsed, commitmenttypes.ErrInvalidProof
+				})
+			},
+			commitmenttypes.ErrInvalidProof,
+		},
+		{
+			"proof height greater than client state latest height",
+			func() {
+				proofHeight = clienttypes.NewHeight(1, 100)
+			},
+			ibcerrors.ErrInvalidHeight,
+		},
+		{
+			"invalid path argument",
+			func() {
+				path = ibcmock.KeyPath{}
+			},
+			ibcerrors.ErrInvalidType,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			suite.SetupWasmWithMockVM()
+
+			endpoint := wasmtesting.NewWasmEndpoint(suite.chainA)
+			err := endpoint.CreateClient()
+			suite.Require().NoError(err)
+
+			path = commitmenttypes.NewMerklePath("/ibc/key/path")
+			proof = []byte("valid proof")
+			proofHeight = clienttypes.NewHeight(0, 1)
+
+			tc.malleate()
+
+			clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), endpoint.ClientID)
```

Comment:
Maybe we can open a PR to remove `suite.ctx`, and `suite.store` so that we don't have two standards

Reply:
Definitely agree, do you wanna open an issue for it :)